core/trino-spi/src/test/java/io/trino/spi/TestSpiBackwardCompatibility.java

```diff
@@ -79,8 +79,9 @@ public void testBackwardIncompatibleEntitiesAreInPreviousSpi()
 
     private static Set<String> getBackwardIncompatibleChanges()
     {
-        String version = new ConnectorContext() {}.getSpiVersion().replace("-SNAPSHOT", "");
-        return BACKWARD_INCOMPATIBLE_CHANGES.get(version);
+        String majorVersion = new ConnectorContext() {}.getSpiVersion()
+                .replaceAll("^(\\d+).*$", "$1");
```

Comment:
Make digit match possessive to emphasize you want "all leading digits" and not "some leading digits"

```suggestion
                .replaceAll("^(\\d++).*$", "$1");
```